config/passport.js

```diff
@@ -99,14 +100,17 @@ module.exports = function(passport, models) {
                     if(!data) {
                         return done(null, false, {info: 'Invalid email or password'});
                     }
-                    if(data.provider==='google') {
+                    if(data.provider === 'google') {
                         return done(null, false, {info: 'Please login with Google'});
                     }
-                    if(data.password!==password) {
-                        return done(null, false, {info: 'Invalid password'});
-                    }
-                    const response = { email: data.email, name: data.name};
-                    return done(null, response);
+                    bcrypt.compare(password, data.password, function(err, response) {
+                        if(response) {
+                            const response = { email: data.email, name: data.name};
+                            return done(null, response);
+                        } else {
```

Comment:
I am not sure if removing else would work well, since node is asynchronous, it might cause issues.